### DISCUSSION TITLE
Support for Gherkin?

### DISCUSSION THREAD

**Discussion by JDRay42** [kuafuai/DevOpsGPT#26] [General]
Is support for the Behavior Driven Development language Gherkin for requirement and test declarations planned?

**Comment by booboosui** [comment]
Good suggestion, we will seriously consider, such as Gherkin specification language, can more clearly define software development requirements and test cases

**Reply by JDRay42** [reply]
From the _Business Analyst Body of Knowledge_ (BABOK), “A requirement is a usable representation of a need. Requirements focus on understanding what kind of value could be delivered if a requirement is fulfilled.”  [Requirements come in many forms](https://www.linkedin.com/pulse/mommy-where-do-requirements-come-from-j-d-ray?utm_source=share&utm_medium=member_ios&utm_campaign=share_via), some of which turn into software development tasks.  Those are “software development requirements”.

Each requirement should come with **acceptance criteria**, which is the answer to the question of “how do we know we got this right?”  A **test case** is a single approach to testing the solution against the acceptance criteria.  Each requirement may have a large number of test cases that must be met to determine that the acceptance criteria are met.

I realize this is a lot to take in all at once, but if you’re advertising an “Enterprise Version” of the software, you’re going to need to address these criteria.

**Reply by booboosui** [reply]
Indeed, as you said, we have also been building standardization and specialization throughout the software development process

**Comment by booboosui** [comment]
I have created an issue about this and thank you for your attention：https://github.com/kuafuai/DevOpsGPT/issues/28
